gammapy/image/models/core.py

```diff
@@ -25,6 +25,21 @@
 
 class SkySpatialModel(Model):
     """Sky spatial model base class."""
+
+    __slots__ = []
+
+    def __init__(self, params):
+        super().__init__(params)
+
+    @property
+    def parameters(self):
+        """Parameters (`~gammapy.utils.modeling.Parameters`)"""
+        return self._parameters
+
+    @parameters.setter
```

Comment:
I think right now there is now real use case, where we want to update the `.parameters` using this setter interface. I'd be slightly in favour of removing it again. If you'd like to keep it maybe add a check, that parameters is an instance of `Parameters`?

Reply:
I added this part in order to fix some test failures. Namely, without this addition line 211 in utils/serialization/xml.py raises `AttributeError: can't set attribute`. Suggestions on alternative ways to fix that?

Reply:
Ok, thanks for the explanation. I just checked the code that you linked and it's definitely
old and partly broken code,  but as it is tested, we should just keep it. So I'm fine to keep the parameter setter.

Reply:
OK. I'll just move it (together with the `@parameters` property) to the `Model` base class, then.